src/com/trollworks/gcs/skill/Skill.java

```diff
@@ -869,7 +869,7 @@ private SkillDefault getBestDefault(SkillDefault excludedDefault) {
                         int level = skillDefault.getType().getSkillLevel(character, skillDefault, excludes);
                         if (skillDefault.getType().isSkillBased()) {
                             String name = skillDefault.getName();
-                            level -= character.getSkillComparedIntegerBonusFor(ID_NAME + ASTERISK, name, skillDefault.getSpecialization());
+                            level -= character.getSkillComparedIntegerBonusFor(ID_NAME + ASTERISK, name, skillDefault.getSpecialization(), "");  //$NON-NLS-1$
```

Comment:
Doesn't this need to know the categories to do the right thing?

Reply:
I was uncertain.    The problem is that SkillDefault did not have a category, so I assumed none.   But I see your point, on line 872, we are trying to find the level of the skill that we are basing this default on... which is different that the Skill from which we are trying to calculate the SkillDefault for.

I think I have it fixed.   It looks up the Skill for the SkillDefault and uses that Skill's categories when asking the character for the correct level for that skill..

Unfortunately, it doesn't correctly update the tooltip :-(    The default is calculated before the Skill and any modifiers that have affected the default are lost.   Not a huge problem (i can already see how to fix it), but I will leave that to the next PR.